actgen/wrappers/remove_redundant_actions.py

```diff
@@ -0,0 +1,39 @@
+import gym
+
+
+class RemoveRedundantActions(gym.Wrapper):
+	"""
+	remove the redundant actions for a gym environment
+	currently, this wrapper is only implemented for MsPacman
+	"""
+	def __init__(self, env):
+		super().__init__(env)
+		if 'MsPacman' not in env.spec.id:
+			raise NotImplementedError('remove redundancy action wrapper is only implemented for MsPacman')
+		self.action_space = gym.spaces.Discrete(5)
+		
+	def step(self, action):
+		if action not in self.action_space:
+			raise RuntimeError("trying to take action not in action space")
+		return self.env.step(action)
+
+
+def test_remove_redundant_actions_env():
+	env = gym.make('MsPacmanNoFrameskip-v4')
+	env = RemoveRedundantActions(env)
+
+	for action in range(5):
+		assert action in env.action_space
+	assert 9 not in env.action_space
+	assert 5 not in env.action_space
+	assert 6 not in env.action_space
+
+	assert env.action_space.n == 5
+
+	assert 0 <= env.action_space.sample() <= 4
```

Comment:
These kinds of randomized test cases worry me a little bit. I'd rather have a for loop that checks each value, rather than relying on randomness.

When you have a randomized test case, it can appear to pass when in fact it sometimes fails, and if we don't notice it fails until a later time, we might incorrectly think some new code is the reason for the failure, instead of knowing it was broken all along.

Reply:
Here, we can probably just check that `0 in env.action_space`, `4 in env.action_space`, and maybe that `env` and `gym.make('MsPacmanNoFrameskip-v4')` have the same action_space type (i.e. `Discrete`).